core/iwasm/libraries/lib-wasi-threads/stress_test/mutex_common.h

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef MUTEX_COMMON_H
+#define MUTEX_COMMON_H
+
+#include <pthread.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <time.h>
+#include <stdlib.h>
+#include <string.h>
+
+enum Constants {
+    NUM_ITER = 250000,
+    NUM_THREADS = 12,
+    NUM_RETRY = 8,
+    RETRY_SLEEP_TIME_US = 1000,
+};
+
+// We're counting how many times each thread was called using this array
+// Main thread is also counted here so we need to make arrays bigger
+typedef struct {
+    int tids[NUM_THREADS + 1];
+    int calls[NUM_THREADS + 1];
+} StatCollector;
+
+typedef struct {
+    pthread_mutex_t *mutex;
+    StatCollector stat;
+    int counter;
+    bool is_sleeping;
+} MutexCounter;
+
+void
+mutex_counter_init(MutexCounter *mutex_counter, pthread_mutex_t *mutex,
+                   bool is_sleeping)
+{
+    mutex_counter->mutex = mutex;
+    memset(&mutex_counter->stat, 0, sizeof(mutex_counter->stat));
+    mutex_counter->counter = 0;
+    mutex_counter->is_sleeping = is_sleeping;
```

Comment:
```suggestion
    memset(mutex_counter, 0, sizeof(*mutex_counter));
    mutex_counter->mutex = mutex;
    mutex_counter->is_sleeping = is_sleeping;
```

not critical though, just a suggestion.